src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveUtility.cs

```diff
@@ -104,9 +104,22 @@ private static Stream GetPackageSignatureFile(
             var buffer = new byte[localFileHeader.UncompressedSize];
 
             reader.BaseStream.Seek(offsetToData, SeekOrigin.Begin);
-#pragma warning disable CA2022 // Avoid inexact read
-            reader.BaseStream.Read(buffer, offset: 0, count: buffer.Length);
-#pragma warning restore CA2022
+#if NET
+            reader.BaseStream.ReadExactly(buffer, offset: 0, count: buffer.Length);
+#else
+            int count = buffer.Length;
+            int offset = 0;
+            while (count > 0)
+            {
+                int read = reader.BaseStream.Read(buffer, offset, count);
+                if (read <= 0)
+                {
+                    throw new EndOfStreamException();
+                }
+                offset += read;
+                count -= read;
+            }
+#endif
```

Comment:
This change seems fine.  With this change, when attempting to read beyond the end of a stream, we'll now throw an `EndOfStreamException`.

For package signing, I think this is OK.  Before the change, either the ZIP metadata is wrong and/or we would've thrown eventually.